scripts/actions/fetch-and-deserialize.js

```diff
@@ -53,63 +53,173 @@ const writeFilesSync = (vfiles) => {
   });
 };
 
-const fetchTranslatedFilesZip = async (fileUris, locale) => {
-  const fileUriStr = fileUris.reduce((str, uri) => {
-    return str.concat(`&fileUris[]=${encodeURIComponent(uri)}`);
-  }, '');
-
-  const localeIdStr = `localeIds[]=${locale}`;
-
-  return fetch(
-    `https://api.smartling.com/files-api/v2/projects/${projectId}/files/zip?${localeIdStr}${fileUriStr}`,
-    {
-      method: 'GET',
-      headers: {
-        Authorization: `Bearer ${await getAccessToken()}`,
-      },
+/**
+ * @param {Object} input
+ * @param {String[]} input.fileUris
+ * @param {Number} batchSize - maximum batch size
+ */
+const createFileUriBatches = ({ fileUris }, batchSize = 50) => {
+  let batches = [];
+  let currentBatch = [];
+
+  for (let i = 0; i < fileUris.length; i++) {
+    currentBatch.push(fileUris[i]);
+
+    if (currentBatch.length === batchSize) {
+      // send the request, reset the count & array
+      batches.push({ fileUris: currentBatch });
+      currentBatch = [];
     }
-  );
-};
+  }
 
-const fetchAndDeserialize = async ({ locale, fileUris }) => {
-  const response = await fetchTranslatedFilesZip(fileUris, locale);
+  // cleanup the last batch
+  if (currentBatch.length != 0) {
+    batches.push({ fileUris: currentBatch });
+  }
 
-  const buffer = await response.buffer();
+  return batches; // [ [{ locale: ja-JP, fileUris: ['hello_world.txt']}], [...], ... ]
```

Comment:
Kris says this structure isn't accurate anymore. But I really appreciate seeing the shape of what we're working with, since it's easy to loose that context through out